jest/jestSetup.js

```diff
@@ -47,6 +47,12 @@ jest.mock('react-native', () => {
       resourceURL:
         'file:///private/var/containers/Bundle/Application/4DCD4D2B-F745-4C70-AD74-8E5F690CF593/ZulipMobile.app/',
     };
+  } else if (ReactNative.Platform.OS === 'android') {
+    ReactNative.NativeModules.TextCompressionModule = {
+      header: 'z|zlib base64|',
+      compress: async (input: string) => `pretend-this-is-compressed:${input}`,
+      decompress: async (input: string) => input.replace('pretend-this-is-compressed:', ''),
```

Comment:
The `header` property and the mock compress/decompress behavior here don't match -- the header is supposed to be the prefix of anything the `compress` method returns, and in `ZulipAsyncStorage.js` we take care not to pass to `decompress` anything that doesn't have the header as a prefix.

A good solution would be to make the header something like `z|mock|` (the `z|…|` syntax is part of the overall protocol as implemented by `ZulipAsyncStorage`), and have `compress` return `` `${header}${input}` ``.

Reply:
Something like this?

```js
    const header = 'z|mock|';
    ReactNative.NativeModules.TextCompressionModule = {
      header,
      compress: async (input: string) => `${header}${input}`,
      decompress: async (input: string) => input.replace(header, ''),
    };
```

I'm looking at `decompress`, in particular. Possibly we don't need to mock that yet, though, since we don't have any tests that use it?